spec/rmt/ssl/certificate_generator_spec.rb

```diff
@@ -131,7 +168,50 @@
         expect(RMT::Execute).to receive(:on_target!).with(
           'openssl', 'req', '-x509', '-new', '-nodes',
           '-key', ssl_files[:ca_private_key], '-sha256', '-days', described_class::OPENSSL_CA_VALIDITY_DAYS,
-          '-out', ssl_files[:ca_certificate], '-config', ssl_files[:ca_config]
+          '-out', ssl_files[:ca_certificate], '-passin', 'stdin', '-config', ssl_files[:ca_config],
+          stdin: ca_password
+        )
+
+        expect(RMT::Execute).to receive(:on_target!).with(
+          'openssl', 'req', '-new', '-key', ssl_files[:server_private_key],
+          '-out', ssl_files[:server_csr], '-config', ssl_files[:server_config]
+        )
+
+        expect(RMT::Execute).to receive(:on_target!).with(
+          'openssl', 'x509', '-req', '-in', ssl_files[:server_csr],
+          '-out', ssl_files[:server_certificate], '-CA', ssl_files[:ca_certificate],
+          '-CAkey', ssl_files[:ca_private_key], '-passin', 'stdin', '-days', described_class::OPENSSL_SERVER_CERT_VALIDITY_DAYS,
+          '-sha256', '-CAcreateserial', '-extensions', 'v3_server_sign',
+          '-extfile', ssl_files[:server_config],
+          stdin: ca_password
+        )
+
+        expect(Yast::SCR).to receive(:Read).with(scr_path, ssl_files[:server_certificate]).and_return(server_cert)
+        expect(Yast::SCR).to receive(:Read).with(scr_path, ssl_files[:ca_certificate]).and_return(ca_cert)
+        expect(Yast::SCR).to receive(:Write).with(scr_path, ssl_files[:server_certificate], server_cert + ca_cert)
+
+        expect(RMT::Execute).to receive(:on_target!).with('chown', 'root:nginx', ssl_files[:ca_certificate])
+        expect(RMT::Execute).to receive(:on_target!).with('chmod', '0640', ssl_files[:ca_certificate])
+
+        generator.generate(common_name, alt_names, ca_password)
+      end
+    end
+
+    context 'when CA is generated without password' do
```

Comment:
Do we actually allow to generate the CA without a password?

Reply:
this is for the case when CA private key is generated before without a password (for example with current version of plugin) but the server certificate is not there. see line "generates only the server certificate"